alica_engine/include/engine/IPlanCreator.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include <memory>
+
+namespace alica
+{
+    class AlicaEngine;
+    class BasicPlan;
+    class Configuration;
+
+    class IPlanCreator
+    {
+    public:
+        virtual ~IPlanCreator() {}
+        virtual std::shared_ptr<BasicPlan> createPlan(int64_t planId) = 0;
```

Comment:
```suggestion
        virtual std::unique_ptr<BasicPlan> createPlan(int64_t planId) const = 0;
```
Factory methods should always use unique pointers to transfer ownership, unless they retain a reference. The client can then choose to transfer to a shared ownership model. Also why wouldn't this be const.